ProcessList.c

```diff
@@ -122,26 +134,165 @@ void ProcessList_remove(ProcessList* this, Process* p) {
 }
 
 Process* ProcessList_get(ProcessList* this, int idx) {
-   return (Process*) (Vector_get(this->processes, idx));
+   return (Process*)Vector_get(this->processes, idx);
 }
 
 int ProcessList_size(ProcessList* this) {
-   return (Vector_size(this->processes));
+   return Vector_size(this->processes);
+}
+
+// ProcessList_updateTreeSetLayer sorts this->displayTreeSet,
+// relying only on itself.
+//
+// Algorithm
+//
+// The algorithm is based on `depth-first search`,
+// even though `breadth-first search` approach may be more efficient on first glance,
+// after comparision it may be not, as it's not save to go deeper without first updating the tree structure.
+// If it would be save that approach would likely bring an advantage in performance.
+//
+// Each call of the function looks for a 'layer'. A 'layer' is a list of processes with the same depth.
+// First it sorts a list. Then it runs the function recursively for each element of the sorted list.
+// After that it updates the settings of processes.
+//
+// It relies on `leftBound` and `rightBound` as an optimization to cut the list size at the time it builds a 'layer'.
+//
+// It uses a temporary Hashtable `draftingTreeSet` because it's not save to traverse a tree
+// and at the same time make changes in it.
```

Comment:
maybe insert another "blank" comment line here so the function name and parameters are a bit more visible?  *shrug*